exchanges/okx/okx_types.go

```diff
@@ -768,46 +756,93 @@ type CurrencyTakerFlow struct {
 
 // UnmarshalJSON deserializes a slice of byte data into CurrencyTakerFlow
 func (c *CurrencyTakerFlow) UnmarshalJSON(data []byte) error {
-	target := [7]any{&c.Timestamp, &c.CallBuyVolume, &c.CallSellVolume, &c.PutBuyVolume, &c.PutSellVolume, &c.CallBlockVolume, &c.PutBlockVolume}
-	return json.Unmarshal(data, &target)
+	return json.Unmarshal(data, &[7]any{&c.Timestamp, &c.CallBuyVolume, &c.CallSellVolume, &c.PutBuyVolume, &c.PutSellVolume, &c.CallBlockVolume, &c.PutBlockVolume})
 }
 
 // PlaceOrderRequestParam requesting parameter for placing an order
 type PlaceOrderRequestParam struct {
 	AssetType     asset.Item `json:"-"`
 	InstrumentID  string     `json:"instId"`
-	TradeMode     string     `json:"tdMode,omitempty"` // cash isolated
+	TradeMode     string     `json:"tdMode"` // cash isolated
 	ClientOrderID string     `json:"clOrdId,omitempty"`
 	Currency      string     `json:"ccy,omitempty"` // Only applicable to cross MARGIN orders in Single-currency margin.
 	OrderTag      string     `json:"tag,omitempty"`
-	Side          string     `json:"side,omitempty"`
-	PositionSide  string     `json:"posSide,omitempty"`
-	OrderType     string     `json:"ordType,omitempty"`
-	Amount        float64    `json:"sz,string,omitempty"`
-	Price         float64    `json:"px,string,omitempty"`
-	ReduceOnly    bool       `json:"reduceOnly,string,omitempty"`
-	QuantityType  string     `json:"tgtCcy,omitempty"` // values base_ccy and quote_ccy
+	Side          string     `json:"side"`
+	PositionSide  string     `json:"posSide,omitempty"` // long/short only for FUTURES and SWAP
+	OrderType     string     `json:"ordType"`           // Time in force for the order
+	Amount        float64    `json:"sz,string"`
+	Price         float64    `json:"px,string,omitempty"` // Only applicable to limit,post_only,fok,ioc,mmp,mmp_and_post_only order.
+	// Options orders
+	PlaceOptionsOrder                    string `json:"pxUsd,omitempty"` // Place options orders in USD
+	PlaceOptionsOrderOnImpliedVolatility string `json:"pxVol,omitempty"` // Place options orders based on implied volatility, where 1 represents 100%
+
+	ReduceOnly              bool   `json:"reduceOnly,string,omitempty"`
+	TargetCurrency          string `json:"tgtCcy,omitempty"`  // values base_ccy and quote_ccy for spot market orders
+	SelfTradePreventionMode string `json:"stpMode,omitempty"` // Default to cancel maker, `cancel_maker`,`cancel_taker`, `cancel_both``
 	// Added in the websocket requests
-	BanAmend   bool       `json:"banAmend,omitempty"` // Whether the SPOT Market Order size can be amended by the system.
-	ExpiryTime types.Time `json:"expTime,omitzero"`
+	BanAmend bool `json:"banAmend,omitempty"` // Whether the SPOT Market Order size can be amended by the system.
+}
+
+// Validate validates the PlaceOrderRequestParam
+func (arg *PlaceOrderRequestParam) Validate() error {
```

Comment:
I feel like this can be moved to okx.go or okx_wrapper instead of important code being inside _types. But I know we do this around the place, so you can click resolve if you want

Reply:
Will leave this unresolved for other people to vote; I don't mind either way